ts41/camel-cased-properties-deep.d.ts

```diff
@@ -0,0 +1,47 @@
+import {CamelCase} from './camel-case';
+
+/**
+Convert object props to camelCase recursively.
+
+This can be useful when, for example, converting some API types from other style.
+
+@see CamelCasedProperties
+@see CamelCase
+@example
+```
+interface User {
+	UserId: number;
+	UserName: string;
+}
+
+interface UserWithFriends {
+	UserInfo: User;
+	UserFriends: User[];
+}
+
+const result: CamelCasedPropertiesDeep<UserWithFriends> = {
+	userInfo: {
+		userId: 1,
+		userName: 'Tom',
+	},
+	userFriends: [
+		{
+			userId: 2,
+			userName: 'Jerry',
+		},
+		{
+			userId: 3,
+			userName: 'Spike',
+		},
+	],
+};
+
+```
+*/
+export type CamelCasedPropertiesDeep<Value> = Value extends Function
```

Comment:
```suggestion
export type CamelCasedPropertiesDeep<ValueType> = ValueType extends Function
```

Reply:
`CamelCased<>` used `Value`, Is it better to keep it same as `CamelCased`?